[pytorch] integrate op dependency analysis process into CI

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30937 [pytorch] integrate op dependency analysis process into CI**
* #31848 [pytorch] update docker image version
* #32027 [pytorch] skip TEST_DILL on Python2
* #32011 [pytorch] gate torch_global_deps with BUILD_SHARED_LIBS flag

Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

ljk53 committed:
[pytorch] integrate op dependency analysis process into CI

Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

[ghstack-poisoned]

ezyang commented on .jenkins/pytorch/build-mobile-code-analysis.sh:
```diff
@@ -0,0 +1,31 @@
+echo "Clang version:"
+clang --version
+
+# !!! THIS HACK SHOULD BE MOVED TO DOCKER BUILD SCRIPT !!!
```
Yes please :)

ljk53 replied:
Can I land this PR for now before moving it to docker image? I will add it to the docker image for sure but it's blocked by the stacked PR for a while... :>

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit a21b442:


* **1/1** broken upstream at merge base 4c34158 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=4c341582ea18f241628c6e1756633bc9f897b76d&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  
* **0/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=a21b44297ba67597109f7f23d9c376602ba8a7aa).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_linux_xenial_py3_clang5_asan_test](https://dr.pytorch.org/build-details.html?build_id=78145529)



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] integrate op dependency analysis process into CI"


Summary:
Custom build and internal build will depend on the analysis result so
let's make sure it doesn't break.

Tested locally with LLVM-5.0, LLVM-7 and LLVM-8.

Test Plan:
- check CI result

Differential Revision: [D18894637](https://our.internmc.facebook.com/intern/diff/D18894637)

[ghstack-poisoned]
